src/app/view/components/_common/dot-menu/dot-menu.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, Input } from '@angular/core';
+import { MenuItem } from 'primeng/primeng';
+import { Observable } from 'rxjs/Observable';
+
+@Component({
+    selector: 'dot-menu',
+    templateUrl: './dot-menu.component.html',
+    styleUrls: ['./dot-menu.component.scss']
+})
+export class DotMenuComponent {
+    @Input() icon: string;
+    @Input() model: MenuItem[];
+    visible = false;
+    constructor() {}
+
+    toggle(): void {
+        this.visible = !this.visible;
+        if (this.visible) {
+            // Skip 1 because the event bubbling capture the document.click
+            Observable.fromEvent(document, 'click')
```

Comment:
You need to unsubscribe when visible is false, because here, you're binding the event a bunch of times.

Reply:
take(1) do that.